modules/claims_api/app/sidekiq/claims_api/v1/poa_form_builder_job.rb

```diff
@@ -47,15 +48,35 @@ def pdf_constructor(poa_code)
       # @param power_of_attorney [ClaimsApi::PowerOfAttorney] Record for this poa change request
       #
       # @return [Hash] All data to be inserted into pdf
-      def data(power_of_attorney)
-        power_of_attorney.form_data.deep_merge({
-                                                 'veteran' => {
-                                                   'firstName' => power_of_attorney.auth_headers['va_eauth_firstName'],
-                                                   'lastName' => power_of_attorney.auth_headers['va_eauth_lastName'],
-                                                   'ssn' => power_of_attorney.auth_headers['va_eauth_pnid'],
-                                                   'birthdate' => power_of_attorney.auth_headers['va_eauth_birthdate']
-                                                 }
-                                               })
+      def data(power_of_attorney, rep)
+        res = power_of_attorney.form_data.deep_merge(
+          {
+            'veteran' => {
+              'firstName' => power_of_attorney.auth_headers['va_eauth_firstName'],
+              'lastName' => power_of_attorney.auth_headers['va_eauth_lastName'],
+              'ssn' => power_of_attorney.auth_headers['va_eauth_pnid'],
+              'birthdate' => power_of_attorney.auth_headers['va_eauth_birthdate']
+            }
+          }
+        )
+        res.merge!({ 'text_signatures' => text_signatures(power_of_attorney, rep) }) unless rep.nil?
+        res
+      end
+
+      def text_signatures(power_of_attorney, rep)
```

Comment:
I think these changes should be in the org builder class. The signatures for 2122 and 2122a are not all in the same location.

Reply:
Or there should be logic to pick the org related signature attributes and or 2122a signature attributes

Reply:
Yes, now that the builder is version there shouldn't be any changes in this file